test/ResultSet/ResultSetIntegrationTest.php

```diff
@@ -90,6 +90,20 @@ public function testCanProvideIteratorAsDataSource()
         $this->assertSame($it, $this->resultSet->getDataSource());
     }
 
+    public function testCanProvideArrayAsDataSource()
+    {
+        $dataSource = [['foo']];
+        $this->resultSet->initialize($dataSource);
+        $this->assertEquals($dataSource[0], (array) $this->resultSet->current());
+
+        $returnType = new ReturnType();
+        $dataSource = [$returnType];
+        $this->resultSet->setArrayObjectPrototype($returnType);
```

Comment:
@bartmcleod the object required for it is an object that need to has `exhangeArray()`, while `ArrayObject` already has it. I tried with ArrayObject and working fine at mine: 
![screen shot 2017-07-05 at 8 02 14 pm](https://user-images.githubusercontent.com/459648/27865558-47789e62-61bd-11e7-8443-5cc136fe9764.png)